tests/unit/test_charm.py

```diff
@@ -28,3 +32,55 @@ def test_flask_pebble_layer(harness: Harness) -> None:
         "user": "flask",
         "group": "flask",
     }
+
+
+@pytest.mark.parametrize(
+    "charm_config, flask_config",
+    [
+        pytest.param({"flask_env": "prod"}, {"env": "prod"}, id="env"),
+        pytest.param({"flask_debug": True}, {"debug": True}, id="debug"),
+        pytest.param({"flask_secret_key": "1234"}, {"secret_key": "1234"}, id="secret_key"),
+        pytest.param(
+            {"flask_preferred_url_scheme": "http"},
+            {"preferred_url_scheme": "HTTP"},
+            id="preferred_url_scheme",
+        ),
+    ],
+)
+def test_charm_state_flask_config(
+    harness: Harness, charm_config: dict, flask_config: dict
+) -> None:
+    """
+    arrange: none
+    act: set flask_* charm configurations.
+    assert: flask_config in the charm state should reflect changes in charm configurations.
+    """
+    harness.begin()
+    harness.update_config(charm_config)
+    charm_state = CharmState.from_charm(harness.charm)
+    assert charm_state.flask_config == flask_config
```

Comment:
If we unittest the charmstate class I think this test is redundant
It's also covered by the integration test.
The only case we could test here is getting the pebble plan and ensure that the environment variables matches, but that's still redundant with the integration test

Reply:
Answering one of the comment about missing unit tests for charm state.
I think this is one of the unit tests for charm state, I will move this to `test_charm_state.py`